jobserv/trigger.py

```diff
@@ -123,6 +130,14 @@ def trigger_build(
     except Exception as e:
         raise _fail_unexpected(b, e)
 
-    trigger_runs(storage, proj_def, b, trigger, params, secrets, None, queue_priority)
-    db.session.commit()
+    def commit_runs(b: Build):
+        trigger_runs(
+            storage, proj_def, b, trigger, params, secrets, None, queue_priority
+        )
+        db.session.commit()
```

Comment:
Just so that I understand the probelm and the fix correctly...

These two lines were taking too long to finish in case of 40+ builds, exceeding the GitHub response timeout.
So the fix is to spawn these runs and update the PR in a separate thread.
Is it right?

Probably need to add a bit more context into the commit message.

Reply:
yep. I'm still testing this. There's no way to properly test this in CI, so I've had to set up gavelci.net with this test code to check it works in the wild